python/GafferSceneUI/LightEditor.py

```diff
@@ -354,6 +382,120 @@ def __editSelectedCells( self, pathListing ) :
+				isinstance( p["value"], Gaffer.BoolPlug )
+			) or isinstance( p, Gaffer.BoolPlug ) for p in plugs
+		) :
+			return False
+
+		currentValues = []
+
+		# Use a single new value for all plugs.
+		# First we need to find out what the new value would be for each plug in isolation.
+		for inspector, pathInspections in inspectors.items() :
+			for path, inspection in pathInspections.items() :
+				currentValue = inspection.value().value if inspection.value() is not None else None
+
+				if isinstance( inspector, GafferSceneUI.Private.AttributeInspector ) :
+					fullAttributes = self.__settingsNode["in"].fullAttributes( path[:-1] )
+					parentValueData = fullAttributes.get( inspector.name(), None )
+					parentValue = parentValueData.value if parentValueData is not None else False
+
+					currentValues.append( currentValue if currentValue is not None else parentValue )
+				else :
+					currentValues.append( currentValue )
+
+		# Now set the value for all plugs, defaulting to `True` if they are not
+		# currently all the same.
+		newValue = not sole( currentValues )
+
+		with Gaffer.UndoScope( self.scriptNode() ) :
+			for inspector, pathInspections in inspectors.items() :
+				for path, inspection in pathInspections.items() :
+					plug = inspection.acquireEdit()
+
+					if isinstance( plug, ( Gaffer.TweakPlug, Gaffer.NameValuePlug ) ) :
+						plug["value"].setValue( newValue )
+						plug["enabled"].setValue( True )
+						if isinstance( plug, Gaffer.TweakPlug ) :
+							plug["mode"].setValue( Gaffer.TweakPlug.Mode.Create )
+					else :
+						plug.setValue( newValue )
+
+		return True
+
+	def __activeInspectionTweaks( self, pathListing ) :
+
+		tweaks = []
+
+		with Gaffer.Context( self.getContext() ) as context :
+			for columnSelection, column in zip( pathListing.getSelection(), pathListing.getColumns() ) :
+				if not isinstance( column, _GafferSceneUI._LightEditorInspectorColumn ) :
+					continue
+				for path in columnSelection.paths() :
+					context["scene:path"] = GafferScene.ScenePlug.stringToPath( path )
+					inspection = column.inspector().inspect()
+					if inspection is not None and inspection.editable() :
+						source = inspection.source()
+						editScope = self.__settingsNode["editScope"].getInput()
+						if (
+							isinstance( source, ( Gaffer.TweakPlug, Gaffer.NameValuePlug ) ) and
+							( editScope is None or editScope.node().isAncestorOf( source ) ) and
+							source["enabled"].getValue()
+						) :
+							tweaks.append( source )
+
+		return tweaks
+
+	def __removableAttributeInspections( self, pathListing ) :
+
+		inspections = []
+
+		with Gaffer.Context( self.getContext() ) as context :
+			for columnSelection, column in zip( pathListing.getSelection(), pathListing.getColumns() ) :
+				if (
+					not isinstance( column, _GafferSceneUI._LightEditorInspectorColumn ) or
+					not isinstance( column.inspector(), GafferSceneUI.Private.AttributeInspector )
```

Comment:
It baffled me that this check was sufficient to stop people trying to delete shader parameters, because ParameterInspector derives form AttributeInspector. It turns out that the reason it works is because there's a bug in the bindings, and we've bound ParameterInspector as a subclass of Inspector and not AttributeInspector.

Reply:
I think I solved half of this in https://github.com/GafferHQ/gaffer/pull/4948/commits/66b80ab6f007e978327ebe4f3712ad9a1a9f8a68 by comparing directly to the `type()` of column.